src/main/sensors/gyro.c

```diff
@@ -448,6 +451,37 @@ FAST_CODE void gyroUpdate(void)
     }
 }
 
+#ifdef USE_SMITH_PREDICTOR
+FAST_CODE_NOINLINE float applySmithPredictor(smithPredictor_t *smithPredictor, float gyroFiltered, int axis) {
+  if (smithPredictor->samples > 1) {
+    smithPredictor->data[smithPredictor->idx] = gyroFiltered;
+    float input = gyroFiltered;
+
+    smithPredictor->idx++;
+    if (smithPredictor->idx > smithPredictor->samples) {
+        smithPredictor->idx = 0;
+    }
+
+    // filter the delayedGyro to help reduce the overall noise this prediction adds
+    float delayedGyro = smithPredictor->data[smithPredictor->idx];
+
+    float delayCompensatedGyro = smithPredictor->smithPredictorStrength * (gyroFiltered - delayedGyro);
+    delayCompensatedGyro = pt1FilterApply(&smithPredictor->smithPredictorFilter, delayCompensatedGyro);
+    gyroFiltered += delayCompensatedGyro;
+
+    if (axis == gyro.gyroDebugAxis) {
```

Comment:
pretty sure this needs to be `    if (axis == (int)(gyro.gyroDebugAxis)) {` to avoid error when `make EXTRA_FLAGS=-Werror all`